@nx/next:custom-server generator broken on Nx v16.3.2

### Current Behavior

The `@nx/next:custom-server` generator does not work on the latest version of Nx. There are multiple issues when used on a freshly generated Next.js app (see "Steps to Reproduce" and "Additional Information" sections).

### Expected Behavior

A freshly generated Next.js app is able to use the `@nx/next:custom-server` generator to add a custom server, and running `npx nx serve my-nextjs-app` starts the custom server without error.

### GitHub Repo

_No response_

### Steps to Reproduce

**Reproduction steps:**
1. Generate a new Nx workspace
    ```
    npx create-nx-workspace@16.3.2
    ``` 
2. Generate a new Next.js app
    ```
    npm install -D @nx/next
    npx nx g @nx/next:init
    npx nx generate @nx/next:application test-nextjs-app --no-interactive
    ```
3. Try to generate a new Next.js custom server
    ```
    npx nx generate @nx/next:custom-server test-nextjs-app --no-interactive
    ```
4. Observe the following error in the terminal:
    ```
     *  Executing task: npx nx generate @nx/next:custom-server test-nextjs-app --no-interactive 


    >  NX  Generating @nx/next:custom-server

    Project test-nextjs-app has invalid config. Did you generate it with "nx g @nx/next:app"?
    ```

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node   : 16.17.1
   OS     : darwin arm64
   npm    : 8.15.0
   Hasher : Native
   
   nx                 : 16.3.2
   @nx/js             : 16.3.2
   @nx/jest           : 16.3.2
   @nx/linter         : 16.3.2
   @nx/workspace      : 16.3.2
   @nx/cypress        : 16.3.2
   @nx/devkit         : 16.3.2
   @nx/eslint-plugin  : 16.3.2
   @nx/next           : 16.3.2
   @nx/react          : 16.3.2
   @nrwl/tao          : 16.3.2
   @nx/web            : 16.3.2
   typescript         : 5.0.4
```


### Failure Logs

_No response_

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

**Additional findings:**
1. The initial error is produced because `@nx/next:custom-server` is looking for a `target.build.options.root` in the Next.js app's `project.json`. Adding `"root": "apps/test-nextjs-app",` allows the generator to execute successfully.
2. Once the custom server has been generated (`npx nx generate @nx/next:custom-server test-nextjs-app --no-interactive`), executing `npx nx run test-nextjs-app:serve` fails because the custom server is not outputted to the correct directory:
    ```
    Error: Cannot find module '/path-to-workspace/test-workspace/dist/apps/test-nextjs-app/main.js' imported from /path-to-workspace/test-workspace/node_modules/@nx/js/src/executors/node/node-with-require-overrides.js
      at new NodeError (node:internal/errors:387:5)
      at finalizeResolution (node:internal/modules/esm/resolve:429:11)
      at moduleResolve (node:internal/modules/esm/resolve:1006:10)
      at defaultResolve (node:internal/modules/esm/resolve:1214:11)
      at nextResolve (node:internal/modules/esm/loader:165:28)
      at ESMLoader.resolve (node:internal/modules/esm/loader:844:30)
      at ESMLoader.getModuleJob (node:internal/modules/esm/loader:431:18)
      at ESMLoader.import (node:internal/modules/esm/loader:528:22)
      at importModuleDynamically (node:internal/modules/cjs/loader:1065:29)
      at importModuleDynamicallyWrapper (node:internal/vm/module:438:21)
    ```
3. The `@nx/next:custom-server` generator puts the custom server in a folder called `server` in the Next.js app: `apps/test-nextjs-app/server/main.ts`. When built, the output is `dist/apps/test-nextjs-app/server/main.js`, but the `serve-custom-server` target is using the `@nx/js:node` executor that is trying to find `main.js` in: `dist/apps/test-nextjs-app/main.js` (note the missing `/server/` directory). The executor does not appear to be configurable to allow specifying a specific directory.
4. Eliminating the `server` folder and moving `main.ts` to the root of the Next.js app can help work around this:
    - Move `apps/test-nextjs-app/server/main.ts` -> `apps/test-nextjs-app/main.ts`
    - Update the `apps/test-nextjs-app/tsconfig.server.json` from `"include": ["server/**/*.ts"]` -> `"include": ["main.ts"]`
    - Update the `build-custom-server` target in the `project.json`: `build-custom-server.options.main` from `apps/test-nextjs-app/server/main.ts` -> `apps/test-nextjs-app/main.ts`
    - Attempt to serve the custom server `npx nx run test-nextjs-app:serve-custom-server` and observe the following error:
        ```
        Error: > Couldn't find any `pages` or `app` directory. Please create one under the project root
            at findPagesDir (/path-to-workspace/test-workspace/node_modules/next/src/lib/find-pages-dir.ts:36:11)
            at DevServer.getRoutes (/path-to-workspace/test-workspace/node_modules/next/src/server/dev/next-dev-server.ts:250:46)
            at new Server (/path-to-workspace/test-workspace/node_modules/next/src/server/base-server.ts:463:41)
            at new NextNodeServer (/path-to-workspace/test-workspace/node_modules/next/src/server/next-server.ts:216:5)
            at new DevServer (/path-to-workspace/test-workspace/node_modules/next/src/server/dev/next-dev-server.ts:188:5)
            at NextServer.createServer (/path-to-workspace/test-workspace/node_modules/next/src/server/next.ts:149:20)
            at /path-to-workspace/test-workspace/node_modules/next/src/server/next.ts:195:34
            at NextServer.prepare (/path-to-workspace/test-workspace/node_modules/next/src/server/next.ts:128:20)
        ```

This was the end of the road for me - I was unable to proceed any further with the workaround.

--- a/packages/next/src/generators/custom-server/custom-server.ts
+++ b/packages/next/src/generators/custom-server/custom-server.ts
@@ -28,7 +28,7 @@ export async function customServerGenerator(
   }
 
   const outputPath = project.targets?.build?.options?.outputPath;
-  const root = project.targets?.build?.options?.root;
+  const root = project.root;
 
   if (
     !root ||

